### PR TITLE
refactoring doc

### PR DESCRIPTION
**Pull request by MalloZup**
Refactoring and make more essential the doc

# read the doc in markdown here
https://github.com/MalloZup/kubic-init/blob/devel-doc/docs/user-guide.md
# todos:

-[ ] add explanation about seeder

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -106,13 +110,13 @@ have combinations of a seeder and regular nodes. For example:
    ```
    This will start a cluster with a _seeder_ and a _node_.
```
Comment (drpaneas) [inline CONTRIBUTING.md]: idea: could you write a few line _what_ is the _seeder_ and what makes it different from any other node?

File: docs/user-guide.md [hunk 2]
```diff
@@ -1,27 +1,115 @@
...
+    source      = "../../init/kubic-init.sysconfig"
+    destination = "/etc/sysconfig/kubic-init"
+
+    source      = "../../init/kubelet-sysctl.conf"
+    destination = "/etc/sysctl.d/99-kubernetes-cri.conf"
+
+* Container runtime
+
+You will need on the `seeder` node,  a container runtime installed. (podman,docker, etc).
+The contaner-runtime will execute the kubic-init container.
```
Comment (drpaneas) [inline docs/user-guide.md]: in the sake of consistency, please pick **one** style and use it along the guide:
- container-runtime
- container run-time
- container runtime

File: docs/user-guide.md [hunk 3]
```diff
@@ -1,27 +1,115 @@
...
+
+On All node pull the `kubic-init`, e.g :
```
Comment (drpaneas) [inline docs/user-guide.md]: if you want to emphasize just make it bold, not starting it with an upper-case letter.
`On **all** node pull the `kubic-init`, e.g:`

also remove the unnecessary spaces (trailing characters at the end of the sentence)

File: docs/user-guide.md [hunk 4]
```diff
@@ -1,27 +1,115 @@
...
+    source      = "../../init/kubelet-sysctl.conf"
+    destination = "/etc/sysctl.d/99-kubernetes-cri.conf"
+
+* Container runtime
+
+You will need on the `seeder` node,  a container runtime installed. (podman,docker, etc).
+The contaner-runtime will execute the kubic-init container.
+
+On All node pull the `kubic-init`, e.g : 
+`docker pull registry.opensuse.org/devel/kubic/containers/container/kubic/init:0.0`
```
Comment (drpaneas) [inline docs/user-guide.md]: This command has to be executed from _within_ the nodes, right? In which step the nodes are getting created though?

File: docs/user-guide.md [hunk 5]
```diff
@@ -1,27 +1,115 @@
...
+
+
+* Kubic-init.yml configuration. (seeder only)
```
Comment (drpaneas) [inline docs/user-guide.md]: Remove the _period_ `.` -- please put period only at the end of the sentences. Also the file is called `kubic-init.yaml` with lower-case `k`.

File: docs/user-guide.md [hunk 6]
```diff
@@ -1,27 +1,115 @@
...
+
+Write the  configuration for kubic-init. into `/etc/kubic/kubic-init.yaml`
```
Comment (drpaneas) [inline docs/user-guide.md]: Remove unnecessary spaces and put periods only at the end of the sentences:

```
- Write the  configuration for kubic-init. into `/etc/kubic/kubic-init.yaml`
+ Write the configuration for kubic-init into `/etc/kubic/kubic-init.yaml`:
```

File: docs/user-guide.md [hunk 7]
```diff
@@ -1,27 +1,115 @@
...
+
+```yaml
+ apiVersion: kubic.suse.com/v1alpha1
+      kind: KubicInitConfiguration
+      network:
+        cni:
+          driver: flannel
+```
+
+For more elaborate configuration have a look at  - [Configuration Reference](#configuration-reference)
```
Comment (drpaneas) [inline docs/user-guide.md]: Fix the spaces.

File: docs/user-guide.md [hunk 8]
```diff
@@ -1,27 +1,115 @@
...
+
+
+* Node Join
+
+To join a node to cluster you will need to use the `token provived by kubic-init
```
Comment (drpaneas) [inline docs/user-guide.md]:
```
- To join a node to cluster you will need to use the `token provived by kubic-init
+ **NOTICE**: To join a node into the cluster you have to use the **token** _provided by_ **kubic-init**.
```

File: docs/user-guide.md [hunk 9]
```diff
@@ -1,27 +1,115 @@
...
+
+
+
+#### Post-installation
+
+After the seeder is ready, you can add manifests for beeing loaded.
```
Comment (drpaneas) [inline docs/user-guide.md]: - when the seeder is ready?
- you add manifest to be loaded? What do you mean by that?

File: docs/user-guide.md [hunk 10]
```diff
@@ -1,27 +1,115 @@
...
+
+- in the seeder node only:
+  - the `kubic-init` container will try to find all the:  `*.yaml` files in
+  - several directories, e.g `/etc/kubic/manifests`
```
Comment (drpaneas) [inline docs/user-guide.md]: is it going to search into other directories as well? If not, then just remove the `e.g.` part. Also the example should be:
`/etc/kubic/manifests/*.yaml`

I would re-write that part into something like that:

- the `kubic-init` container tries to find all the _manifests_ listed in `/etc/kubic/manifests/*.yaml` 

but this is up to you :)

File: docs/user-guide.md [hunk 11]
```diff
@@ -1,27 +1,115 @@
...
+  They will be loaded and treated as [Go templates](https://golang.org/pkg/text/template/),
```
Comment (drpaneas) [inline docs/user-guide.md]: This information is not related on how to install the product. This is how the product works, which should be in another page in the documentation. Putting it here it just confuses the reader -- thinking that he has to do something about finding files and stuff like that. I would advise to remove this section from this page and put it somewhere else.

File: docs/user-guide.md [hunk 12]
```diff
@@ -1,27 +1,115 @@
...
+  `kubic-init` process will gather the manifest file from that URL. 
+  - finally, `kubic-init` process will create or update the resources loaded.
+
+
+___
+# Configuration reference
+
+There is an example [kubic-init.yaml](../config/kubic-init.yaml) that has all the possible configurations
+
+# Features explained in details.
```
Comment (drpaneas) [inline docs/user-guide.md]: Do not put period in the end of the sentences if they are headers.

File: docs/user-guide.md~ [hunk 13]
```diff
@@ -0,0 +1,74 @@
+# Kubic-init User Guide:
```
Comment (drpaneas) [inline docs/user-guide.md~]: What is this file? I see you have `docs/user-guide.md~` and also `docs/user-guide.md`. Is this by mistake?

File: docs/user-guide.md~ [hunk 14]
```diff
@@ -0,0 +1,74 @@
...
+
+- [Installation](#installation)
+- [Configuration](#configuration)
+- [concepts](#CONCEPTS)
+
+# INSTALLATION
+
+THIS section contains instructions for installing Kubic-init in development and production environments.
```
Comment (drpaneas) [inline docs/user-guide.md~]: `THIS` -> `This`

File: docs/user-guide.md~ [hunk 15]
```diff
@@ -0,0 +1,74 @@
+# Kubic-init User Guide:
+
+- [Installation](#installation)
+- [Configuration](#configuration)
+- [concepts](#CONCEPTS)
+
+# INSTALLATION
```
Comment (drpaneas) [inline docs/user-guide.md~]: `# INSTALLATION` -> `# Installation`

File: docs/user-guide.md~ [hunk 16]
```diff
@@ -0,0 +1,74 @@
...
+
+THIS section contains instructions for installing Kubic-init in development and production environments.
+
+- Install a cluster with kubic-init with manual configuration.
+
+* [Preparing the bootstrap](config-pre.md)
+* BLABLA
```
Comment (drpaneas) [inline docs/user-guide.md~]: Put `TODO` instead of `BLABLA` so we know what to look for.

File: CONTRIBUTING.md [hunk 17]
```diff
@@ -92,7 +96,7 @@ cluster with the help of Terraform. All these targets have some common
 steps, like:
 
   * starting `kubic`-based VMs
-  * generating some config files from the [`cloud-init` templates](../deployments/cloud-init)
+  * generating some config files from the [`cloud-init` templates](deployments/cloud-init)
```
Comment (drpaneas) [inline CONTRIBUTING.md]: Remove `some`

File: config/manifests/README.md [hunk 18]
```diff
@@ -1,7 +1,7 @@
 # Post-control plane manifests files
 
 - (**on build**),
-  - files in this directory will be copied to `/usr/lib/kubic/manifests` in
+  - files in this directory will be copied to `usr/lib/kubic/manifests/` in
```
Comment (drpaneas) [inline config/manifests/README.md]: I guess you mean: `/usr/lib/kubic/manifests/`

File: docs/config-pre.md [hunk 19]
```diff
@@ -1,4 +1,7 @@
-# Configuring the Kubernetes bootstrap
 
-* Adding manifests for being loaded [after the control-plane is ready](../config/manifests/README.md).
+h
```
Comment (drpaneas) [inline docs/config-pre.md]: ?????

File: docs/user-guide.md [hunk 20]
```diff
@@ -1,27 +1,115 @@
...
+You also need to satisfy `kubeadm` Requirements, avaible at:
+https://kubernetes.io/docs/setup/independent/install-kubeadm/#before-you-begin
+
+OS-Requrirements: this guide is based on kubic-images, but with some modification/adaptation works on other OS.
+
+#### Installation
+
+##### Seeder Node and others Nodes configuration:
+
+Copy the  config files needed for configuring `kubelet` and `kubic-init`
```
[the remark this answers is not in the document]
Comment (drpaneas) [reply docs/user-guide.md]: Also why the user has to copy files by hand? Don't we have `kubic-init` already [packaged](https://build.opensuse.org/package/show/devel%3Akubic/kubic-init)? I would expect that the user installs the RPM and all the configuration is handled by the RPM mechanism.

File: docs/user-guide.md~ [hunk 21]
```diff
@@ -0,0 +1,74 @@
...
+
+Inside we have the section registries: this section will let you configure mirrors for registries
+
+* multiple mirrors can be set
+
+* each prefix can have multiple mirrors addresses
+
+* each mirror can have certificates configured. This will be used for security to validate the origin of the server.
+
+> Certificates: In this configuration you must add the Certificate content, the Fingerprint and the Hash Algorithm that was used.
```
Comment (drpaneas) [inline docs/user-guide.md~]: do not use upper-case letters to emphasize -- but use markdown semantics, like **bold**. e.g.
```
- Certificate
+ **certificate**
```

File: docs/user-guide.md~ [hunk 22]
```diff
@@ -0,0 +1,74 @@
...
+This is useful for air-gapped environments, where it is not possible to access upstream registries and you have configured a local mirror.
+
+In this scenario, you have to configure the daemon.json file for the container engine to be able to find the initial images, otherwise Kubic would not start if this was not configured.
+
+Example:
+
+```yaml
+bootstrap:
+  registries:
+    - prefix: https://registry.suse.com
```
Comment (drpaneas) [inline docs/user-guide.md~]: is this an example you have tested? If so please re-test it and put the information for the **opensuse registry** instead of the **suse** one.

File: docs/user-guide.md~ [hunk 23]
```diff
@@ -0,0 +1,74 @@
...
+
+# Configuration
+
+There is an example [kubic-init.yaml](../config/kubic-init.yaml) that has all the possible configurations for kubic.
+
+# Features:
+
+#### Bootstrap
+
+The bootstrap section is reserved to the configuration that is needed in actions that take part before initializing kubic.
```
Comment (drpaneas) [inline docs/user-guide.md~]: I guess you mean `kubic-init` instead of `kubic`.

File: docs/user-guide.md~ [hunk 24]
```diff
@@ -0,0 +1,74 @@
...
+
+* [Preparing the bootstrap](config-pre.md)
+* BLABLA
+
+
+
+
+# Configuration
+
+There is an example [kubic-init.yaml](../config/kubic-init.yaml) that has all the possible configurations for kubic.
```
Comment (drpaneas) [inline docs/user-guide.md~]: I guess you mean `kubic-init` instead of `kubic`.

File: docs/user-guide.md~ [hunk 25]
```diff
@@ -0,0 +1,74 @@
+# Kubic-init User Guide:
+
+- [Installation](#installation)
+- [Configuration](#configuration)
+- [concepts](#CONCEPTS)
```
Comment (drpaneas) [inline docs/user-guide.md~]: For the shake of consistency please use one style of letter-case and do not mix them.

File: docs/user-guide.md [hunk 26]
```diff
@@ -1,27 +1,115 @@
...
+- [Configuration Reference](#configuration-reference)
 
-### Configuring Kubic before bootstrapping the cluster
+# Installation
 
-* [Preparing the bootstrap](config-pre.md)
-* Deployments:
-  * [Deployment examples](../deployments/README.md)
-  * Using [`cloud-init`](../deployments/cloud-init/README.md).
+This section contains instructions for installing Kubic-init in development and production environments.
```
[the remark this answers is not in the document]
Comment (MalloZup) [reply docs/user-guide.md]: yop ,  good point fixing

File: docs/user-guide.md [hunk 27]
```diff
@@ -1,27 +1,115 @@
...
 
-* [External authentication](https://github.com/kubic-project/dex-operator/blob/master/README.md)
-* [Adding private Docker registries](https://github.com/kubic-project/registries-operator/blob/master/README.md)
+# Manual Install
 
-### The configuration file
+## Install a cluster with kubic-init with manual configuration.
 
-There is an example [kubic-init.yaml](../config/kubic-init.yaml) that has all the possible configurations for kubic.
+This installation will guide you to install 1 seeder and 1 node.
```
[the remark this answers is not in the document]
Comment (MalloZup) [reply docs/user-guide.md]: No, :smile: . The goal of the installation is to provide the minimal stuff.

There will be a guideLine for using HA like having 2 master with X nodes.

Currently with kubic-init you can have 1 master and XX nodes. But this is due to kubeadm.

File: docs/user-guide.md [hunk 28]
```diff
@@ -1,27 +1,115 @@
...
-### Post control-plane configuration
+# Terraform based deployment:
 
-The Kubernetes cluster running on Kubic can be
-configured
+LINK to deployment directory.
 
-* [External authentication](https://github.com/kubic-project/dex-operator/blob/master/README.md)
-* [Adding private Docker registries](https://github.com/kubic-project/registries-operator/blob/master/README.md)
+# Manual Install
```
[the remark this answers is not in the document]
Comment (MalloZup) [reply docs/user-guide.md]: yop thx

File: docs/user-guide.md [hunk 29]
```diff
@@ -1,27 +1,115 @@
...
+This installation will guide you to install 1 seeder and 1 node.
 
-#### Bootstrap
+* Configuring the Kubernetes bootstrap
+
+You can read about the concept of [bootstrap](docs/design/design-bootstrap.md) with kubic-init.
+
+#### Requirements for Nodes (seeder and others)
+
+* kubelet
```
[the remark this answers is not in the document]
Comment (MalloZup) [reply docs/user-guide.md]: yop the thing is i don't want to put OS specific commands. If ubuntu or other distros want to install kubic-init we should not provide specific OS commands.

Is install xx enough rmo your POV?

File: docs/user-guide.md~ (hunk 22 above)
Comment (MalloZup) [reply docs/user-guide.md~]: @drpaneas this documentation was added by @chentex . i dont have added it. I agree with that. but is out of scope of this pr. I dunno if airgap feature work with opensuse it should afaik

File: docs/user-guide.md~ (hunk 25 above)
Comment (MalloZup) [reply docs/user-guide.md~]: yop

File: docs/user-guide.md [hunk 30]
```diff
@@ -1,27 +1,115 @@
...
-  * [Deployment examples](../deployments/README.md)
-  * Using [`cloud-init`](../deployments/cloud-init/README.md).
+This section contains instructions for installing Kubic-init in development and production environments.
 
-### Post control-plane configuration
+# Terraform based deployment:
 
-The Kubernetes cluster running on Kubic can be
-configured
+LINK to deployment directory.
```
[the remark this answers is not in the document]
Comment (MalloZup) [reply docs/user-guide.md]: done

File: docs/user-guide.md [hunk 31]
```diff
@@ -1,27 +1,115 @@
...
+
+#### Requirements for Nodes (seeder and others)
+
+* kubelet
+* kubectl
+
+You also need to satisfy `kubeadm` Requirements, avaible at:
+https://kubernetes.io/docs/setup/independent/install-kubeadm/#before-you-begin
+
+OS-Requrirements: this guide is based on kubic-images, but with some modification/adaptation works on other OS.
```
[the remark this answers is not in the document]
Comment (MalloZup) [reply docs/user-guide.md]: fixed

File: docs/user-guide.md (hunk 20 above)
[the remark this answers is not in the document]
Comment (MalloZup) [reply docs/user-guide.md]: done

File: docs/user-guide.md (hunk 4 above)
Comment (MalloZup) [reply docs/user-guide.md]: we assume nodes are created already

**Comment by MalloZup** [comment]
@inercia @ereslibre @nirmoy can you review this https://github.com/MalloZup/kubic-init/blob/devel-doc/docs/user-guide.md only this? 

I know at moment isn't optimal but is better then before.

Imho we can merge this, and then we could improve it later on.  Or i will re-adress other comments 
 
tia :rocket: :guitar:

File: CONTRIBUTING.md [hunk 32]
```diff
@@ -1,4 +1,14 @@
...
+
+  - [Building](#building)
+  - [Deploy](#deploy)
+  - [Testing](#testing)
+
+# Building
+
+A simple `make` should be enough. This should compile [the main
+function](cmd/kubic-init/main.go) and generate a `kubic-init` binary as
+well as a _Docker_ image.
```
Comment (nirmoy) [inline CONTRIBUTING.md]: I am not sure if `make` also generate/update docker image.

Comment (MalloZup) [reply CONTRIBUTING.md]: yop, but we state `should` :smile: . i will check out how we can improve. thx :rocket:

Comment (inercia) [reply CONTRIBUTING.md]: I didn't enable the automatic build of the docker image because it slowed down the usual workflow. I would skip this part in this paragraph.

File: docs/user-guide.md [hunk 33]
```diff
@@ -1,29 +1,101 @@
...
+
+Here is an example of minimal configuration.
+
+```yaml
+ apiVersion: kubic.suse.com/v1alpha1
+      kind: KubicInitConfiguration
+      network:
+        cni:
+          driver: flannel
+```
```
Comment (inercia) [inline docs/user-guide.md]: Is this YAML properly formatted? Maybe it is just a display error here in the PR...

File: docs/user-guide.md [hunk 34]
```diff
@@ -1,29 +1,101 @@
...
+
+`curl https://raw.githubusercontent.com/kubic-project/kubic-init/master/init/kubic-init.systemd.conf > /etc/systemd/system/kubic-init.service`
+`systemctl enable --now kubic-init`
+
+
+For more elaborate configuration have a look at  - [Configuration Reference](#configuration-reference)
+
+* Node Join
+
+To join a node to cluster you will need to use the `token provived by kubic-init
```
Comment (inercia) [inline docs/user-guide.md]:
```suggestion
To join a node to cluster you will need to use the `token` provived by kubic-init
```

File: docs/user-guide.md [hunk 35]
```diff
@@ -1,29 +1,101 @@
...
+
+___
+
+
+# Configuration reference
+
+There is an example [kubic-init.yaml](../config/kubic-init.yaml) that has all the possible configurations
```
Comment (inercia) [inline docs/user-guide.md]:
```suggestion
There is an example [kubic-init.yaml](../config/kubic-init.yaml.sample) that has all the possible configurations
```

File: docs/user-guide.md (hunk 33 above)
Comment (MalloZup) [reply docs/user-guide.md]: yop fixing

**Comment by MalloZup** [comment]
merging is better then before, not enough yet but still ok.

Thank you for all review
